dask/dataframe/multi.py

```diff
@@ -261,8 +260,7 @@ def concat_indexed_dataframes(dfs, join='outer'):
                for df, empty in zip(part, empties)]
               for part in parts]
 
-    token = tokenize(([df._name for df in dfs], join))
-    name = 'concat-indexed-' + token
+    name = 'concat-indexed-' + tokenize(join, *dfs)
```

Comment:
Thanks for making `tokenize` smarter.  This feels a lot more sensible than the way it was before.